data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py

```diff
@@ -0,0 +1,167 @@
+        f = open(files[i], "r")
+        lines = f.read().splitlines()
+        f.close()
+        runTimes = ast.literal_eval(lines[0]) 
+        wallSeconds = [item[1] for item in runTimes]
+        cpuSeconds = [item[0] for item in runTimes]
+        addPlotData(dataWall, fileNames[i], wallSeconds, xSteps)
+        addPlotData(dataCPU, fileNames[i], cpuSeconds, xSteps)
+    return dataWall, dataCPU
+
+def getPlotColors():
+    colorScheme = "Dark2"
+    cmap = get_cmap(colorScheme)  # type: matplotlib.colors.ListedColormap
+    return cmap.colors 
+
+def setBarPlotLegends(data, colors):
+    colorCounter = 0
+    legendColors = {}
+    for keyName in data.keys():
+        legendColors[keyName] = colors[colorCounter]
+        colorCounter += 1      
+    labels = list(legendColors.keys())
+    handles = [plt.Rectangle((0,0),1,1, color=legendColors[label]) for label in labels]
+    plt.legend(handles, labels)
+
+def setLinePlotLegends(data, colors):
+    counter = 0
+    legendColors = {}
+    legendMarkers = {}
+    for keyName in data.keys():
+        legendColors[keyName] = colors[counter]
+        legendMarkers[keyName] = markers[counter]
+        counter += 1      
+    labels = list(legendColors.keys())
+    handles = [plt.Line2D([0,0],[0,1], color=legendColors[label], marker=legendMarkers[label], linestyle='-') for label in labels]
+    plt.legend(handles, labels)
+
+def setDescriptiveText(minExponent, titleName):
+    plt.suptitle(titleName, fontsize='xx-large')
+    plt.xlabel('Number of PSM Lines x$10^{}$'.format(str(len(str(minExponent))-1)), fontsize='x-large')
+    plt.ylabel('Time in Seconds', fontsize='x-large')
+
+def getBarChart(data):
+    colors = getPlotColors()
+    df = pd.DataFrame(data)
+
+    ax = df.plot(kind='bar', width=0.4, color=colors, zorder=3)
+    ax.grid(zorder=0)
+
+    setBarPlotLegends(data, colors)
+    plt.setp(plt.xticks()[1], rotation=0)
+    return plt
+
+def renderLines(data, colors):
+    dataFrames = []
+    for keyName in data.keys():
+        tmpData = {keyName: data[keyName]}
+        print(tmpData)
+        dataFrames.append(pd.DataFrame(tmpData))
+    prevAx = None
+    counter = 0
+    for df in dataFrames:
+        if prevAx is not None:
+            prevAx = df.plot(color=colors[counter], marker=markers[counter], ax=prevAx)
+        else:
+            prevAx = df.plot(color=colors[counter], marker=markers[counter])
+        counter = (counter + 1) % min(len(markers), len(colors))
+
+def formatHorizontalLine(data):
+    ax = plt.gca()
+    ax.xaxis.set_major_formatter(FormatStrFormatter('%g'))
+    numXBins = len(data[next(iter(data))])
+    plt.locator_params(axis = 'x', nbins = numXBins)
+
+def getLineChart(data):
+    colors = getPlotColors()
+    renderLines(data, colors)
+
+    setLinePlotLegends(data, colors)
+    plt.setp(plt.xticks()[1], rotation=0)
+    plt.grid(True)
+
+    formatHorizontalLine(data)
+    return plt
+
+def getFiles():
+    files = sys.argv[1:]
```

Comment:
It would be a bit nicer to pass this array in as an argument from the top-level, instead of hard-coding the function to use sys.argv as the source.

Reply:
Probably, but I am not sure if it is worth spending time on this since it is a very simple function.